include/deal.II/lac/full_matrix.h

```diff
@@ -995,12 +995,42 @@ class FullMatrix : public Table<2, number>
    * @param A The first matrix (m x n).
    * @param B The second matrix (p x q).
    * @param adding If `true`, the result is added to the current matrix. If
-   *               `false` (default), the current matrix is overwritten.
+   *               `false`, the current matrix is overwritten.
+   *
+   * Note that if `adding` is not provided, the function below is called
+   * resulting in `adding` being `false`.
+   */
+  void
+  kronecker_product(const FullMatrix<number> &A,
+                    const FullMatrix<number> &B,
+                    const bool                adding);
+
+
+  /**
+   * Compute the Kronecker product of multiple FullMatrix objects.
+   *
+   * This variadic overload computes \f$A_1 \otimes A_2 \otimes \cdots \otimes
+   * A_k\f$ by performing a left-associative fold: first \f$T_2 := A_1 \otimes
+   * A_2\f$, then \f$T_3 := T_2 \otimes A_3\f$, and so on until all factors are
+   * consumed. The resulting matrix has size \f[ \left(\prod_{i=1}^k
+   * \mathrm{rows}(A_i)\right) \times \left(\prod_{i=1}^k
+   * \mathrm{cols}(A_i)\right). \f]
+   *
+   * All arguments must be FullMatrix<number> objects.
+   * For example, to compute \f$A_1 \otimes A_2 \otimes A_3\f$ call
+   * @code
+   * result.kronecker_product(A_1, A_2, A_3);
+   * @endcode
+   *
+   * The result overwrites the current object. The order of the input arguments
+   * is significant: \f$A_1\f$ is the left-most factor in the Kronecker product.
    */
+  template <typename... Rest>
   void
   kronecker_product(const FullMatrix<number> &A,
                     const FullMatrix<number> &B,
```

Comment:
Actually, I meant for you to rename these two arguments to `A_1, A_2` as well, to stay consistent with the documentation above.